recipes/libcoap/all/conanfile.py

```diff
@@ -86,12 +88,21 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        pkgconfig_filename = "libcoap-2{}".format("-{}".format(self.options.dtls_backend) if self.options.dtls_backend else "")
-        self.cpp_info.names["pkg_config"] = pkgconfig_filename 
+        library_name = ""
+        pkgconfig_name = ""
+        if self.version == "cci.20200424":
+            library_name = "coap"
+            pkgconfig_name = "libcoap-2"
+        else:
+            library_name = "coap-3"
+            pkgconfig_name = "libcoap-3"
+
         self.cpp_info.components["coap"].names["cmake_find_package"] = "coap"
         self.cpp_info.components["coap"].names["cmake_find_package_multi"] = "coap"
-        self.cpp_info.components["coap"].names["pkg_config"] = pkgconfig_filename
-        self.cpp_info.components["coap"].libs = ["coap"]
+        pkgconfig_filename = "{}{}".format(pkgconfig_name, "-{}".format(self.options.dtls_backend) if self.options.dtls_backend else "")
+        self.cpp_info.names["pkg_config"] = pkgconfig_filename
```

Comment:
It looks like I have managed to remove one line setting the pkgconfig_file. In the original recipe both these lines where present. Did not realize there where two places this information where set.

self.cpp_info.names["pkg_config"] = pkgconfig_filename 
self.cpp_info.components["coap"].names["pkg_config"] = pkgconfig_filename

I assume I should re-add the missing one, please advice.

Reply:
You should only need one and the component one should be used :) 

Good catch!

Reply:
It's now fixed.